pkg/util/proxy/proxy.go

```diff
@@ -159,12 +168,32 @@ func Location(cluster *clusterapis.Cluster, tlsConfig *tls.Config) (*url.URL, ht
 		return nil, nil, err
 	}
 
-	proxyTransport, err := createProxyTransport(cluster, tlsConfig)
+	if v, ok := clusterEndpointInfoStore.Load(cluster.UID); ok {
+		clusterEndpointsInfo := v.(clusterEndpointInfo)
+		return location, clusterEndpointsInfo.Transport, nil
+	}
+
+	endpointInfo, err, _ := singleExecution.Do(string(cluster.UID), func() (interface{}, error) {
+		if value, ok := clusterEndpointInfoStore.Load(cluster.UID); ok {
+			return value, nil
+		}
+		proxyTransport, err := createProxyTransport(cluster, tlsConfig)
+		if err != nil {
+			return nil, err
+		}
+		clusterEndpoint := clusterEndpointInfo{
+			Transport: proxyTransport,
+		}
+		clusterEndpointInfoStore.Store(cluster.UID, clusterEndpointInfo{
```

Comment:
Just a question: When will the `clusterEndpointInfo` be removed from the `clusterEndpointInfoStore`?

Reply:
+1